tests/test_multichannel.py

```diff
@@ -59,6 +92,25 @@ def test_stft_multi(y_multi):
     assert not np.allclose(D0, D1)
 
 
+def test_onset_strength(y_multi):
+
+    # Verify that a stereo spectral flux onset strength envelope matches on
+    # each channel individually
+    y, sr = y_multi
+
+    D = librosa.onset.onset_strength(y)
+
+    D0 = librosa.onset.onset_strength(y[0])
+    D1 = librosa.onset.onset_strength(y[1])
+
+    # Check each channel
+    assert np.allclose(D[0], D0, atol=1e-1, rtol=1e-1) # test in dB units
+    assert np.allclose(D[1], D1, atol=1e-1, rtol=1e-1) # test in dB units
+
+    # Check that they're not both the same
+    assert not np.allclose(D0, D1, atol=1e-1, rtol=1e-1) # test in dB units
```

Comment:
These tolerances are way loose.  How far off are the results actually?  (Also I don't think they're in dB)

Reply:
mmhmm, then there's something I do not understand here: 
https://github.com/librosa/librosa/blob/fad5538330312635a7ac91a9b9bb92d078b16593/librosa/onset.py#L571

Here are some general stats on the comparison being made during the test.

```
np.abs(D[0] - D0)
min: 0.0
max: 0.14240837
mean: 0.008507246

np.abs(D[1] - D1)
min: 0.0
max: 1.4305115e-06
mean: 7.587927e-08


np.abs(D0 - D1)
min: 0.0
max: 0.44862747
mean: 0.100642346
```

Reply:
Ah, right.  This is another instance of the `power_to_db` calculation being sensitive to all channels simultaneously, as noted here: https://github.com/librosa/librosa/pull/1351#issuecomment-894711524

TLDR: it truncates to 80dB below peak, but the peak finder operates across all dimensions.  (I think this is correct behavior, and what we want most of the time.)

The fix here, I think, would be to use spectrogram input directly instead of implicitly computing a melspec by providing the time-domain input.